### PR TITLE
Changes to generator/util/gen*.py files

### PR DESCRIPTION
**Pull request by mattagape**

### CODE REVIEW

File: generator/util/generateCode.py [hunk 1]
```diff
@@ -374,76 +378,93 @@ def generate_other_library_code_files(name, ob):
...
+        directories = populate_other_library_directories(pkgname, lang)
     print('creating directory structure for {0}'.format(pkgname))
     index = 0
     all_present = False
     if os.path.exists(pkgname):
-        all_present = check_directory_structure()
+        all_present = check_directory_structure(directories)
         index = 1
-    done = True
+    done = True  # TODO `ok` prob more accurate than `done`, esp. in this loop!
```
Comment (skeating) [inline generator/util/generateCode.py]: I agree - its definitely not 'done' when it does this

File: generator/util/generateCode.py [hunk 2]
```diff
@@ -456,25 +477,35 @@ def create_dir_structure(pkgname, lang, overwrite):
...
     for directory in directories:
         if not os.path.exists(directory):
             return False
     return True
 
 
 def create_dir(name, skip_existing):
-    """ skip_existing seems to be a synonym for `overwrite` parameter, used
-     in function create_dir_structure(). I don't think it is being used properly here.
+    """ skip_existing seems to be a synonym for `overwrite` parameter,
```
Comment (skeating) [inline generator/util/generateCode.py]: I agree skip_existing == overwrite

File: generator/util/generateCode.py [hunk 3]
```diff
@@ -70,35 +63,40 @@ def generate_code_for(filename, overwrite=True):
...
-                = global_variables.return_codes['parsing error']
-    if global_variables.code_returned == \
-            global_variables.return_codes['success']:
+        except Exception:
+            gv.code_returned = gv.return_codes['parsing error']
+    if gv.code_returned == gv.return_codes['success']:
         name = ob['name'.lower()]
-        language = global_variables.language
+        language = gv.language
         # TO DO REMEMBER TO REMOVE   TODO why?
```
Comment (skeating) [inline generator/util/generateCode.py]: I think my note to remove it relates to the fact that it should be a try/catch but I removed it when I was testing and should add it back if doing a release.

**Review by skeating — APPROVED** [review]

File: generator/util/generateCode.py (hunk 1 above)
Comment (mattagape) [reply generator/util/generateCode.py]: Changed.

File: generator/util/generateCode.py (hunk 3 above)
Comment (mattagape) [reply generator/util/generateCode.py]: So @skeating , should I add it back in?

File: generator/util/generateCode.py (hunk 2 above)
Comment (mattagape) [reply generator/util/generateCode.py]: Now fixed.

File: generator/util/generateCode.py (hunk 3 above)
Comment (mattagape) [reply generator/util/generateCode.py]: As agreed, I have uncommented the try/catch block and all tests continued to work, so I'm going to mark this as resolved and then merge this PR.

**Pull request merged**